packages/devextreme/js/core/utils/date.js

```diff
@@ -689,6 +689,45 @@ const getMachineTimezoneName = () => {
         : null;
 };
 
+const getRangesByDates = (dates) => {
+    const datesInMilliseconds = dates.map((value) => {
+        const date = new Date(new Date(value).setHours(0, 0, 0, 0)).getTime();
+
+        return date;
+    });
+
+    const sortedDates = datesInMilliseconds.sort((a, b) => a - b);
+
+    const getRange = (date, dates, index) => {
+        const range = date === dates[index - 1]
+            ? [date]
+            : [date, dates[index - 1]];
+
+        return range.map((value) => new Date(value));
```

Comment:
It’s safer to use the [dateSerialization.deserializeDate(value)](https://github.com/marker-dao/DevExtreme/blob/24_2_chat_messagesend_timeout/packages/devextreme/js/__internal/ui/calendar/m_calendar.ts#L255-L256) function to convert many possible formats of the incoming value to a Date format

Reply:
I cannot explain from top of my head why it's safer but yeah, probably we should use this solution at least for a consistency

Reply:
[e6875c297c1d4e3c571476fa8c008d6410276286]